pkg/collect/host_network.go

```diff
@@ -20,13 +23,67 @@ const (
 	NetworkStatusConnected            = "connected"
 	NetworkStatusErrorOther           = "error"
 	NetworkStatusBindPermissionDenied = "bind-permission-denied"
+	NetworkStatusInvalidAddress       = "invalid-address"
 )
 
 type NetworkStatusResult struct {
-	Status NetworkStatus `json:"status"`
+	Status  NetworkStatus `json:"status"`
+	Message string        `json:"message"`
+}
+
+func isValidIPCandidate(address string) bool {
+	validChars := "0123456789."
+	for i := range address {
+		if !strings.Contains(validChars, string(address[i])) {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidLoadBalancerAddress(address string) bool {
+	splitString := strings.Split(address, ":")
+
+	if len(splitString) != 2 { // should be hostAddress:port
+		return false
+	}
+	hostAddress := splitString[0]
+	port, err := strconv.Atoi(splitString[1])
+	if err != nil {
+		return false
+	}
+	portErrors := validation.IsValidPortNum(port)
+
+	if len(portErrors) > 0 {
+		return false
+	}
+
+	// Checking for uppercase letters
+	if strings.ToLower(hostAddress) != hostAddress {
+		return false
+	}
+
+	// Checking if it's all numbers and .
+	if isValidIPCandidate(hostAddress) {
```

Comment:
I don't think you need this candidate function. You can do
```
if len(validation.IsValidIP(hostAddress)) == 0 {
  return true
}
```
and then go on to validate as hostname if not.

Reply:
That sounds like a great idea!

Reply:
Forgot that without that validation check it would actually fail the unit tests we created.

I created a branch and PR with the suggested changes [here]([https://github.com/replicatedhq/troubleshoot/pull/388).

Basically the k8s api validation check is too loose.

Reply:
In that case what about using `net.ParseIP`?

Reply:
The issue is that https://github.com/replicatedhq/troubleshoot/blob/feature/validate-tcp-load-balancer-address/pkg/collect/host_network.go#L76 returns cases of false positives. 

If it goes to that function it will pass if it contains all digits and '.' characters. The only valid configuration for that is if it is an IP address. So that's why the inside of the IF statement returns a fail if it's not an IP but should be.